node/tests/connection_shutdown_test.rs

```diff
@@ -10,11 +10,12 @@ use std::net::{IpAddr, TcpListener, TcpStream};
 use std::net::{Shutdown, SocketAddr};
 use std::str::FromStr;
 use std::time::Duration;
-use std::{io, thread};
+use std::{env, io, thread};
 
 // 'node' below must not be named '_' alone or disappear, or the MASQNode will be immediately reclaimed.
 #[test]
 fn proxy_client_stream_reader_dies_when_client_stream_is_killed_integration() {
+    env::set_var("MASQ_INTEGRATION_TEST", "true");
```

Comment:
There is a possibility of making use of the guard we have in the test utils, it is placed like this:

```rust
 let _guard = EnvironmentGuard::new();
```

to the first line of a test. Its purpose is to keep other tests away from being biased by something this test sets on. When this test ends, it will automatically invalidate all changes made to the environment.

If its visibility allows its use in integration tests (or you consider moving it over to a common place), you might want to give it a try also here.